plugins/processors/t128_filter/t128_filter.go

```diff
@@ -0,0 +1,148 @@
+package t128_filter
+
+import (
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/plugins/processors"
+)
+
+const sampleConfig = `
+[[processors.t128_filter]]
+  ## The conditions that must be met to pass a metric through. This is similar
+  ## behavior to a tagpass, but the multiple tags are ANDed
+  [[processors.t128_filter.condition]]
+
+  [processors.t128_filter.condition.tags]
+     #tag1 = ["value1", "value2"]
+	 #tag2 = ["value3"]
+
+  [[processors.t128_filter.condition]]
+
+  [processors.t128_filter.condition.tags]
+     #tag1 = ["value3"]
+`
+
+type tags map[string][]string
+
+type Condition struct {
+	Tags tags `toml:"tags"`
+}
+
+type T128Filter struct {
+	Conditions []Condition `toml:"condition"`
+
+	log     telegraf.Logger `toml:"-"`
+	matcher matcher         `toml:"-"`
+}
+
+func (r *T128Filter) SampleConfig() string {
+	return sampleConfig
+}
+
+func (r *T128Filter) Description() string {
+	return "Filter metrics from being emitted."
+}
+
+type matcher interface {
+	Matches(telegraf.Metric) bool
+}
+
+type exactMatcher struct {
+	tag    string
+	values []string
+}
+
+func (m exactMatcher) Matches(point telegraf.Metric) bool {
+	value, ok := point.GetTag(m.tag)
+	if !ok {
+		return false
+	}
+
+	for _, expectedValue := range m.values {
+		if value == expectedValue {
+			return true
+		}
+	}
+
+	return false
+}
+
+type andConjMatcher struct {
+	matchers []matcher
+}
+
+func (c andConjMatcher) Matches(point telegraf.Metric) bool {
+	for _, matcher := range c.matchers {
+		if !matcher.Matches(point) {
+			return false
+		}
+	}
+
+	return true
+}
+
+type orConjMatcher struct {
+	matchers []matcher
+}
+
+func (c orConjMatcher) Matches(point telegraf.Metric) bool {
+	for _, matcher := range c.matchers {
+		if matcher.Matches(point) {
+			return true
+		}
+	}
+
+	return false
+}
+
+func createMatcher(conditions []Condition) (matcher, error) {
+	conditionMatchers := make([]matcher, len(conditions))
+	for i, condition := range conditions {
+		tagMatchers := getTagMatchers(condition.Tags)
+
+		conditionMatchers[i] = andConjMatcher{matchers: tagMatchers}
+	}
+
+	return orConjMatcher{conditionMatchers}, nil
+}
+
+func getTagMatchers(tags tags) []matcher {
+	tagMatchers := make([]matcher, len(tags))
+
+	j := 0
+	for tagKey, tagValues := range tags {
+		tagMatchers[j] = exactMatcher{tag: tagKey, values: tagValues}
+		j++
+	}
+
+	return tagMatchers
+}
+
+func (r *T128Filter) Apply(in ...telegraf.Metric) []telegraf.Metric {
+	filteredPoints := make([]telegraf.Metric, 0)
+
+	for _, point := range in {
+		if r.matcher.Matches(point) {
+			filteredPoints = append(filteredPoints, point)
```

Comment:
An alternative to creating a whole new slice and appending each (could end up not even filtering anything) is to set the fields to an empty map. Telegraf should take care of removing those then as it's an invalid line.

Reply:
I looked into this as well as a `Drop` method on a metric which I think would mark it for dropping by telegraf. I'd need to test that out and figure out how to test that, but that also might be a good option. I created another task to do that to keep things moving.